rllib/algorithms/algorithm_config.py

```diff
@@ -462,13 +461,13 @@ def __init__(self, algo_class=None):
         self.worker_restore_timeout_s = 1800
 
         # `self.rl_module()`
-        self.rl_module_spec = None
-        self._enable_rl_module_api = False
+        self._rl_module_spec = None
```

Comment:
Made this private (plus a @property for `rl_module_spec`). We should never automatically (or inside `validate()`) set any properties. This now makes a lot of tests better as they don't require to magically wait for things to change under the hood after calling `validate()`.